src/peft/peft_model.py

```diff
@@ -125,6 +125,23 @@ def __init__(self, model: PreTrainedModel, peft_config: PeftConfig, adapter_name
         if hasattr(self.base_model, "config") and hasattr(self.base_model.config, "pretraining_tp"):
             self.base_model.config.pretraining_tp = 1
 
+    @property
+    def _ddp_params_and_buffers_to_ignore(self) -> set[str]:
+        # Add an attribute _ddp_params_and_buffers_to_ignore. This is sometimes required for DDP, as we may
+        # want to ignore certain parameters to avoid the error "Parameters which did not receive grad for rank X".
+        # See issue 899.
+        ddp_params_and_buffers_to_ignore = set()
+        for name, module in self.named_modules():
+            if module is self:
+                # avoid infinite recursion
+                continue
+
+            module_params_and_buffers_to_ignore = getattr(module, "_ddp_params_and_buffers_to_ignore", [])
```

Comment:
```suggestion
            if isinstance(module, BaseTunerLayer):
                module_params_and_buffers_to_ignore = module._ddp_params_and_buffers_to_ignore
```

Then inside the if block perform the for loop